source/navitiacommon/navitiacommon/models.py

```diff
@@ -401,6 +401,12 @@ def get_from_id_or_name(cls, id=None, name=None):
         else:
             raise Exception({'error': 'instance is required'}, 400)
 
+    def delete_dataset(self):
+        jobs = db.session.query(Job).filter(Job.instance_id == self.id).all()
+        for j in jobs:
+            db.session.delete(j)
+            db.session.commit()
```

Comment:
```python
for j in jobs:
    db.session.delete(j)
db.session.commit()
```

Reply:
you can probably do this with less code
Job.delete().where(Job.instance_id == self.id)
not sure, to be tested

Reply:
Done